leveldb/db_compaction.go

```diff
@@ -844,6 +844,11 @@ func (db *DB) tCompaction() {
 			switch cmd := x.(type) {
 			case cAuto:
 				if cmd.ackC != nil {
+					// Check the write pause state before caching it.
+					if db.resumeWrite() {
+						x.ack(nil)
+						continue
```

Comment:
Don't use `continue` here, use `if else` instead. If compaction need to happen then let it happen. Thanks!

Reply:
Updated.